File: download_images.py
#!/usr/bin/env python3
"""
HyperFit Product Images Auto-Downloader
Downloads all 20 product images to correct folders
"""

import os
import requests
from pathlib import Path
import time

# Image URLs mapping
IMAGES = {
    'supplements': {
        'on-gold-whey.jpg': 'https://m.media-amazon.com/images/I/71gNiWfR4IL._AC_SY879_.jpg',
        'myprotein-whey.jpg': 'https://images.myprotein.com/mblogs/content/uploads/2021/05/23100605/NEW-PRODUCT-LAUNCH.jpg',
        'on-creatine.jpg': 'https://m.media-amazon.com/images/I/71kKmWqmyHL._AC_SY879_.jpg',
        'c4-preworkout.jpg': 'https://m.media-amazon.com/images/I/91kq5qbT1QL._AC_SY879_.jpg',
        'bcaa.jpg': 'https://m.media-amazon.com/images/I/812WOqH30qL._AC_SY879_.jpg',
        'on-isolate.jpg': 'https://m.media-amazon.com/images/I/81LvLpEyVVL._AC_SY879_.jpg',
        'on-casein.jpg': 'https://m.media-amazon.com/images/I/71L8k5FEwWL._AC_SY879_.jpg',
    },
    'apparel': {
        'gymshark-joggers.jpg': 'https://images.gymshark.com/files/cdn/v1659009471/cms/c_crop,h_600,w_500,x_0,y_0/6b51fc6f-9e5e-44a5-b7b1-f635e959bda2.jpeg',
        'gymshark-tank.jpg': 'https://images.gymshark.com/files/cdn/v1659108836/cms/c_crop,h_600,w_500,x_0,y_0/d22e6d12-2c2c-4a9f-bbea-1abb18e14c77.jpeg',
        'lululemon-pants.jpg': 'https://images.lululemon.com/c_crop,h_600,w_500,x_0,y_0/f/collab_img/abc-pants-tall-warp-knit-black/LW9APTS_0_BHEATH.jpg',
        'ua-compression.jpg': 'https://images.underarmour.com/is/image/Underarmour/SS21_1327033_001_00_1280x960.jpg',
        'nike-shorts.jpg': 'https://images.nike.com/is/image/DotCom_PDP_PHON_800x800_1600/CJ2158_010_A_PREM.jpg',
    },
    'equipment': {
        'rogue-bar.jpg': 'https://www.roguefitness.com/media/catalog/product/cache/1/image/650x650/e9c3970ab036de1ea8c930bead280715/e/c/echo_bar_1_4.jpg',
        'adj-dumbbells.jpg': 'https://www.roguefitness.com/media/catalog/product/cache/1/image/650x650/e9c3970ab036de1ea8c930bead280715/a/d/adi_adjustable_dumbbells_5_50_1.jpg',
        'yoga-mat.jpg': 'https://images-cdn.ubuy.co.in/634f7a44880fc50d0b0e33f2-liforme-yoga-mat-yoga-mats-amazon.jpg',
        'kettlebell.jpg': 'https://m.media-amazon.com/images/I/71a8p8vB-6L._AC_SY879_.jpg',
        'foam-roller.jpg': 'https://images-na.ssl-images-amazon.com/images/I/81f5G7s9z4L._AC_SY879_.jpg',
    },
    'accessories': {
        'schiek-straps.jpg': 'https://images-na.ssl-images-amazon.com/images/I/51vb0lI1mIL._AC_SY879_.jpg',
        'sbd-belt.jpg': 'https://cdn.shopify.com/s/files/1/0023/0102/8844/products/ipf-belt-2lb-_1_1024x1024.jpg',
        'gripgenie.jpg': 'https://m.media-amazon.com/images/I/61e2OG8a6QL._AC_SY879_.jpg',
    }
}

def download_images():
    """Download all product images"""

    # Get base path (frontend public/images)
    base_path = Path(__file__).parent / 'frontend' / 'public' / 'images'

    total = 0
    success = 0
    failed = 0

    print("🚀 Starting HyperFit Product Image Download...\n")

    for category, images in IMAGES.items():
        category_path = base_path / category

        print(f"\n📁 {category.upper()} ({len(images)} images)")
        print("=" * 50)

        for filename, url in images.items():
            total += 1
            file_path = category_path / filename

            try:
                print(f"  ⏳ Downloading: {filename}...", end=" ")

                # Download with timeout and headers
                headers = {
                    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
                }
                response = requests.get(url, timeout=10, headers=headers, stream=True)
                response.raise_for_status()

                # Save file
                with open(file_path, 'wb') as f:
                    for chunk in response.iter_content(chunk_size=8192):
                        if chunk:
                            f.write(chunk)

                file_size = os.path.getsize(file_path) / 1024  # KB
                print(f"✅ ({file_size:.1f} KB)")
                success += 1

            except Exception as e:
                print(f"❌ ERROR: {str(e)}")
                failed += 1
                # Create placeholder if download fails
                with open(file_path, 'wb') as f:
                    f.write(b'placeholder')

            time.sleep(0.5)  # Be respectful to servers

    # Summary
    print("\n" + "=" * 50)
    print("\n📊 DOWNLOAD SUMMARY")
    print(f"  Total: {total}")
    print(f"  ✅ Success: {success}")
    print(f"  ❌ Failed: {failed}")
    print(f"\n✨ Images saved to: {base_path}")
    print("\n🎉 Ready for Phase 2: Premium UI Upgrade!")

if __name__ == '__main__':
    try:
        download_images()
    except KeyboardInterrupt:
        print("\n\n⚠️  Download cancelled by user")
    except Exception as e:
        print(f"\n\n❌ Error: {e}")

File: download-images.js
const fs = require("fs");
const path = require("path");
const https = require("https");

// Using Unsplash URLs which are reliable and designed for downloads
const IMAGES = {
  supplements: {
    "on-gold-whey.jpg": "https://images.unsplash.com/photo-1577401132379-0e3c6a7bb41a?w=600&h=600&fit=crop",
    "myprotein-whey.jpg": "https://images.unsplash.com/photo-1596267633876-4d18d4e35000?w=600&h=600&fit=crop",
    "on-creatine.jpg": "https://images.unsplash.com/photo-1607623814075-e51df1bdc82f?w=600&h=600&fit=crop",
    "c4-preworkout.jpg": "https://images.unsplash.com/photo-1609895318053-82ad36b7a4e9?w=600&h=600&fit=crop",
    "bcaa.jpg": "https://images.unsplash.com/photo-1599599810694-b5ac4dd3c2c3?w=600&h=600&fit=crop",
    "on-isolate.jpg": "https://images.unsplash.com/photo-1577401132379-0e3c6a7bb41a?w=600&h=600&fit=crop",
    "on-casein.jpg": "https://images.unsplash.com/photo-1596267633876-4d18d4e35000?w=600&h=600&fit=crop",
  },
  apparel: {
    "gymshark-joggers.jpg": "https://images.unsplash.com/photo-1506629082955-511b1aa562c8?w=600&h=600&fit=crop",
    "gymshark-tank.jpg": "https://images.unsplash.com/photo-1521572163474-6864f9cf17ab?w=600&h=600&fit=crop",
    "lululemon-pants.jpg": "https://images.unsplash.com/photo-1552886030-3b4a4d84a9e6?w=600&h=600&fit=crop",
    "ua-compression.jpg": "https://images.unsplash.com/photo-1521572163474-6864f9cf17ab?w=600&h=600&fit=crop",
    "nike-shorts.jpg": "https://images.unsplash.com/photo-1539185441755-769473a23570?w=600&h=600&fit=crop",
  },
  equipment: {
    "rogue-bar.jpg": "https://images.unsplash.com/photo-1534438327276-14e5300c3a48?w=600&h=600&fit=crop",
    "adj-dumbbells.jpg": "https://images.unsplash.com/photo-1638536532686-d610adfc8e5c?w=600&h=600&fit=crop",
    "yoga-mat.jpg": "https://images.unsplash.com/photo-1601925260368-ae2f83cf8b7f?w=600&h=600&fit=crop",
    "kettlebell.jpg": "https://images.unsplash.com/photo-1534438327276-14e5300c3a48?w=600&h=600&fit=crop",
    "foam-roller.jpg": "https://images.unsplash.com/photo-1518611505868-48510c2e2e38?w=600&h=600&fit=crop",
  },
  accessories: {
    "schiek-straps.jpg": "https://images.unsplash.com/photo-1583454110551-21f2fa2afe61?w=600&h=600&fit=crop",
    "sbd-belt.jpg": "https://images.unsplash.com/photo-1605296867304-46d5465a25f1?w=600&h=600&fit=crop",
    "gripgenie.jpg": "https://images.unsplash.com/photo-1517836357463-d25ddfcbf042?w=600&h=600&fit=crop",
  }
};

async function downloadImage(url, filepath) {
  return new Promise((resolve, reject) => {
    https.get(url, { headers: { "User-Agent": "Mozilla/5.0" } }, (response) => {
      if (response.statusCode === 200) {
        const file = fs.createWriteStream(filepath);
        response.pipe(file);
        file.on("finish", () => {
          file.close();
          resolve();
        });
        file.on("error", reject);
      } else {
        reject(new Error(`HTTP ${response.statusCode}`));
      }
    }).on("error", reject);
  });
}

async function downloadAllImages() {
  const basePath = path.join(__dirname, "frontend", "public", "images");

  let total = 0;
  let success = 0;
  let failed = 0;

  console.log("\n🚀 Downloading HyperFit Product Images (Unsplash - Reliable)...\n");

  for (const [category, images] of Object.entries(IMAGES)) {
    const categoryPath = path.join(basePath, category);

    console.log(`\n📁 ${category.toUpperCase()} (${Object.keys(images).length} images)`);
    console.log("=".repeat(50));

    for (const [filename, url] of Object.entries(images)) {
      total++;
      const filepath = path.join(categoryPath, filename);

      try {
        process.stdout.write(`  ⏳ Downloading: ${filename}... `);
        await downloadImage(url, filepath);

        const stats = fs.statSync(filepath);
        const sizeKB = (stats.size / 1024).toFixed(1);
        console.log(`✅ (${sizeKB} KB)`);
        success++;
      } catch (error) {
        console.log(`❌ ERROR: ${error.message}`);
        failed++;
      }

      await new Promise((resolve) => setTimeout(resolve, 500));
    }
  }

  console.log("\n" + "=".repeat(50));
  console.log("\n📊 DOWNLOAD SUMMARY");
  console.log(`  Total: ${total}`);
  console.log(`  ✅ Success: ${success}`);
  console.log(`  ❌ Failed: ${failed}`);
  console.log(`\n✨ Images saved to: ${basePath}`);
  console.log("\n✅ Phase 1 Complete! Ready for Phase 2: Premium UI Upgrade!\n");
}

downloadAllImages().catch((error) => {
  console.error("\n❌ Error:", error.message);
  process.exit(1);
});
